### PR TITLE
fix printing message for thrown non-error objects

### LINKED ISSUE #757: Thrown strings are not displayed
### Version

0.1.4

### Platform

Darwin home.home 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:22 PDT 2022; root:xnu-8020.121.3~4/RELEASE_X86_64 x86_64

### What steps will reproduce the bug?

```js
throw "badness 10000";
```

### How often does it reproduce? Is there a required condition?

always

### What is the expected behavior?

print out the actual error string

### What do you see instead?

nothing

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by alexkuz**
Fixes #757 
(stack trace/code still needs to be printed)

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
Can you include a screenshot of what it looks like when printed?

**Comment by alexkuz** [comment]
should also cover other cases probably
| Before | After | Node |
|----|----|----|
| image | image | image |

**Comment by alexkuz** [comment]
But at least it covers the case in the issue

**Comment by Jarred-Sumner** [comment]
Thanks

It should say `error:` at the beginning with the same formatting instead of only the string label. It is not clear that there was an error

**Comment by alexkuz** [comment]
Yeah, Deno does it more consistently, I think it's clearer this way:
image

**Comment by Jarred-Sumner** [comment]
"Uncaught" has a more precise meaning - that there was an uncaught exception. This error handler can run in cases where it wasn't uncaught. For example `Bun.inspect(myErrorInstance)` will return the error formatted as a string.

We do need to add a promise rejection handler and probably exit(1) on promise rejection, but that is slightly different

**Comment by alexkuz** [comment]
This better?
image

**Comment by alexkuz** [comment]
oh, it should still print "error" for other cases

**Comment by alexkuz** [comment]
image

**Comment by Jarred-Sumner** [comment]
For `Date()` and `{}` case, it would be best to call the equivalent of `.toString()` (`getZigString()` should do it) so that it shows some text instead of no text

Merging in current state anyway since this is strictly better than present state

File: src/bun.js/javascript.zig [hunk 1]
```diff
@@ -2074,40 +2044,55 @@ pub const VirtualMachine = struct {
...
 
-        if (add_extra_line) writer.writeAll("\n") catch unreachable;
+        if (add_extra_line) try writer.writeAll("\n");
 
         try printStackTrace(@TypeOf(writer), writer, exception.stack, allow_ansi_color);
     }
+
+    fn printErrorNameAndMessage(_: *VirtualMachine, name: ZigString, message: ZigString, comptime Writer: type, writer: Writer, comptime allow_ansi_color: bool) !void {
+        if (name.len > 0 and message.len > 0) {
+            try writer.print(comptime Output.prettyFmt("<r><red>{s}<r><d>:<r> <b>{s}<r>\n", allow_ansi_color), .{
```
Comment (Jarred-Sumner) [inline src/bun.js/javascript.zig:2084]: The formatting specifier here should be `{any}` instead of `{s}` I think, but that may have already been incorrect

**Comment by Jarred-Sumner** [comment]
Thanks

**Pull request merged**